planner/core/exhaust_physical_plans.go

```diff
@@ -1091,20 +1113,12 @@ func (p *LogicalJoin) tryToGetIndexJoin(prop *property.PhysicalProperty) (indexJ
 		rhsCardinality := p.Children()[1].statsInfo().Count()
 
 		leftJoins := p.getIndexJoinByOuterIdx(prop, 0)
-		if leftJoins != nil && leftOuter && !rightOuter {
-			return leftJoins, true
-		}
-
-		rightJoins := p.getIndexJoinByOuterIdx(prop, 1)
-		if rightJoins != nil && rightOuter && !leftOuter {
-			return rightJoins, true
-		}
-
-		if leftJoins != nil && lhsCardinality < rhsCardinality {
```

Comment:
We should remove this heuristic rule actually? Previously, cost of index join computed only depends on row count of outer child, then this rule can hold, but now we have changed the cost computation of index join to consider inner row count as well, so it would not always be better if outer row count is smaller?

Reply:
Should we make this change in an individual PR?
This PR only refines the code structure.

Reply:
Sure.